tests/transactions/tests.py

```diff
@@ -378,32 +378,35 @@ def test_implicit_savepoint_rollback(self):
         Reporter.objects.create(id=1)
         Reporter.objects.create(id=2)
 
-        other_thread_ready = threading.Event()
+        main_thread_ready = threading.Event()
 
         def other_thread():
             try:
                 with transaction.atomic():
-                    list(Reporter.objects.filter(id=1))
-                    other_thread_ready.set()
-                    # We cannot synchronize the two threads with an event here
-                    # because the main thread locks. Sleep for a little while.
-                    time.sleep(1)
-                    # 2) ... and this line deadlocks. (see below for 1)
+                    list(Reporter.objects.filter(id=1).select_for_update())
+                    main_thread_ready.wait()
+                    # 1) This line locks... (see below for 2)
                     Reporter.objects.exclude(id=1).update(id=2)
+            except Exception as e:
+                print ("in other thread:", e, file=sys.stderr)
+                raise
```

Comment:
This `except:` clause was a debugging aid of course, I'm not sure it needs to be removed as when the tests run correctly, no exception is raised here.